File: Python/2418.py
# 2418. Sort the People
# https://leetcode.com/problems/sort-the-people/

class Solution:
  def sortPeople(self, names: List[str], heights: List[int]) -> List[str]:
    zipped = sorted(zip(heights, names), reverse=True, key=lambda v: v[0])
    return list(list(zip(*zipped))[1])

File: Rust/2418.rs
/**
 * 2418. Sort the People
 * https://leetcode.com/problems/sort-the-people/
 */

use std::iter::zip;

impl Solution {
  pub fn sort_people(names: Vec<String>, heights: Vec<i32>) -> Vec<String> {
    let mut zipped: Vec<(i32, String)> = zip(heights, names).collect();
    zipped.sort_unstable_by(|a, b| b.cmp(a));
    return zipped
      .into_iter()
      .map(|(_, n)| n)
      .collect::<Vec<String>>();
  }
}
